shop4All-backend/src/main/java/com/example/shop4All_backend/services/UserService.java

```diff
@@ -19,15 +21,18 @@ public class UserService {
 
     @Autowired
     private PasswordEncoder passwordEncoder;
+    @Autowired
+    private CategoryRepo categoryRepo;
 
     //seller registration
     public User registerNewSeller(User seller) {
+        validateAllCredentialsSeller(seller);
         validateEmail(seller.getUserEmail());
         validatePassword(seller.getUserPassword());
```

Comment:
I think you can do the following:
1) Rename validateAllCredentialsSellers method with validateCredentials.
2) The parameter for this method will be some Generic Type (T). Depending of the type of object (Seller or Buyer) do the specific logic for you entity.
3) You can also add the validateEmail() and validatePassword into the validateCredentials method.

This way you can avoid the code duplication.